yjit.c

```diff
@@ -1160,6 +1160,7 @@ static void
 yjit_root_free(void *ptr)
 {
     // Do nothing. The root lives as long as the process.
```

Comment:
👋 Does this comment need to be removed/updated not that the function is no longer "do nothing"?

Reply:
I'm indifferent. It's still technically "doing nothing" (i.e. not freeing any YJIT memory) but `xfree(ptr)` is required for correctness. If you allocate memory to a TypedData object (which it does using `TypedData_Make_Struct`), then you have to free it.

I also realized that we can just use `RUBY_DEFAULT_FREE` instead, so this function is completely removed.